libs/sdk-backend-base/src/toolkit/execution.ts

```diff
@@ -55,4 +62,57 @@ export abstract class AbstractExecutionFactory implements IExecutionFactory {
 
         return this.forDefinition(def);
     }
+
+    public forInsightByRef(insight: IInsight, filters?: IFilter[]): IPreparedExecution {
+        return this.forInsight(insight, filters);
+    }
+}
+
+/**
+ * This implementation of execution factory allows transparent injection of fixed set of filters to all
```

Comment:
Maybe we should mention that this is a concat and not the custom logic we use in InsightView for example.